src/vs/platform/windows/electron-main/windowImpl.ts

```diff
@@ -200,8 +199,10 @@ export class CodeWindow extends Disposable implements ICodeWindow {
 			let useSandbox = false;
 			if (typeof windowSettings?.experimental?.useSandbox === 'boolean') {
 				useSandbox = windowSettings.experimental.useSandbox;
+			} else if (this.productService.quality === 'stable' && this.stateMainService.getItem<boolean>('window.experimental.useSandbox', false)) {
+				useSandbox = true;
```

Comment:
We use state service as another source for sandbox but ensure to first check any setting by the user in case user has made an explicit choice.